File: src/fib.js
function fibonacci(n) {
  let a = 1;
  let b = 1;

  while (n-- > 1) {
    let t = a;
    a = b;
    b += t;
  }

  return b;
}

function benchmark(num) {
  var last = 0;
  for (var i = 0; i < num; i++) {
    last = fibonacci(i);
  }

  return last;
}

if (typeof module !== "undefined" && module.exports) {
  module.exports = {
    fibonacci,
    benchmark
  };
}

File: src/fib.c
#include <webassembly.h>

export double fibonacci(double n) {
  double a = 1;
  double b = 1;

  while (n-- > 1) {
    double t = a;
    a = b;
    b += t;
  }

  return b;
}

export double benchmark(double num) {
  double last = 0;
  for (double i = 0; i < num; i++) {
    last = fibonacci(i);
  }

  return last;
}
